### PR TITLE
Migrate to VS17 and update dependencies

### PR DESCRIPTION
**Pull request by ColinSullivan1**
Move to a more current build environment and update the .NET core dependencies to current versions.
 
* Change project files to VS 2017 format
* Move to NET 2.0 dependencies
* Fix a bug found closing a connection
* Fix a test failing due to test parameters

### CODE REVIEW

File: NATS.Client/Conn.cs [hunk 1]
```diff
@@ -3564,7 +3564,13 @@ private void close(ConnState closeState, bool invokeDelegates)
                             bw.Flush();
                         }
                         catch (Exception) { /* ignore */ }
-                        bw.Dispose();
+
+						try
```
Comment (kozlovic) [inline NATS.Client/Conn.cs]: tabs?

**Review by kozlovic — APPROVED** [review]
LGTM

File: NATS.Client/Conn.cs (hunk 1 above)
Comment (ColinSullivan1) [reply NATS.Client/Conn.cs]: Thanks Ivan - will fix.

**Comment by cjbhaines** [comment]
@ColinSullivan1 can we put the NATS.Client Library back to .NET Standard 1.6 please? By moving forward to 2.0 we have reduced the comparability and forces all consumers to be on 2.0. The best practice guide is to target the lowest possible version of .NET Standard. I just tried this locally and it still builds ok. This still gains the benefits of the new SDK.

**Comment by ColinSullivan1** [comment]
..True.  I'll move it back.

**Comment by cjbhaines** [comment]
Thanks a lot!

**Pull request merged**